src/query/parse.ts

```diff
@@ -153,11 +153,20 @@ export const QUERY_LANGUAGE = P.createLanguage<QueryLanguageTypes>({
                         return P.succeed({ type });
                     case "calendar":
                         return P.whitespace.then(
-                            P.seqMap(q.namedField, field => {
+                            P.seqMap(
+                                q.namedField,
+                                P.whitespace
+                                    .then(P.regexp(/DEFAULT\s+MONTH/i))
+                                    .then(P.whitespace)
+                                    .then(EXPRESSION.date)
+                                    .skip(P.optWhitespace)
+                                    .atMost(1),
+                                (field, month) => {
                                 return {
                                     type,
                                     showId: true,
                                     field,
+                                    displayedMonth: month.length == 1 ? month[0] : undefined,
                                 } as QueryHeader;
                             })
```

Comment:
Optional Super Nitpick (NP): it helps improve readability by adding another tab there. I.e.

```js
P.seqMap(
    q.namedField,
    P.whitespace
        .then(P.regexp(/DEFAULT\s+MONTH/i))
        .then(P.whitespace)
        .then(EXPRESSION.date)
        .skip(P.optWhitespace)
        .atMost(1),
    (field, month) => {
        return {
            type,
            showId: true,
            field,
            displayedMonth: month.length == 1 ? month[0] : undefined,
        } as QueryHeader;
    }
)
```